hack/tools/fetch-images/main.go

```diff
@@ -0,0 +1,290 @@
+		os.Exit(1)
+	}
+	addonImages, err := getImagesForAddons(helmChartConfigMap, chartDirectory)
+	if err != nil {
+		fmt.Println("failed to get images from addons", err.Error())
+		os.Exit(1)
+	}
+	images = append(images, addonImages...)
+	images = slices.Compact(images)
+	for _, image := range images {
+		fmt.Println(image)
+	}
+}
+
+func EnsureFullPath(filename string) (string, error) {
+	if path.IsAbs(filename) {
+		return filename, nil
+	}
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to get wd: %w", err)
+	}
+	fullPath := path.Join(wd, filename)
+	fullPath = path.Clean(fullPath)
+	_, err = os.Stat(fullPath)
+	if err != nil {
+		return "", err
+	}
+	return fullPath, nil
+}
+
+type HelmChartFromConfigMap struct {
+	Version    string `yaml:"ChartVersion"`
+	Repository string `yaml:"RepositoryURL"`
+	ChartName  string `yaml:"ChartName"`
+}
+
+func getImagesForAddons(helmChartConfigMap, carenChartDirectory string) ([]string, error) {
+	templatedConfigMap, err := getTemplatedHelmConfigMap(helmChartConfigMap)
+	if err != nil {
+		return nil, err
+	}
+	cm := &corev1.ConfigMap{}
+	err = yaml.Unmarshal([]byte(templatedConfigMap), cm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal configmap to object %w", err)
+	}
+	images := []string{}
+	for _, chartInfoRaw := range cm.Data {
+		var settings HelmChartFromConfigMap
+		err = yamlv2.Unmarshal([]byte(chartInfoRaw), &settings)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal chart info from configmap %w", err)
+		}
+		info := &ChartInfo{
+			name: settings.ChartName,
+			repo: settings.Repository,
+		}
+		valuesFile := getValuesFileForChartIfNeeded(settings.ChartName, carenChartDirectory)
+		if valuesFile != "" {
+			info.valuesFile = valuesFile
+		}
+		if settings.ChartName == "aws-cloud-controller-manager" {
+			values, err := getHelmValues(carenChartDirectory)
+			if err != nil {
+				return nil, err
+			}
+			awsImages, found, err := unstructured.NestedStringMap(values, "hooks", "ccm", "aws", "k8sMinorVersionToCCMVersion")
+			if !found {
+				return images, fmt.Errorf("failed to find k8sMinorVersionToCCMVersion from file %s",
+					path.Join(carenChartDirectory, "values.yaml"))
+			}
+			if err != nil {
+				return images, fmt.Errorf("failed to get map k8sMinorVersionToCCMVersion with error %w",
+					err)
+			}
+			for _, tag := range awsImages {
+				info.stringValues = []string{
+					fmt.Sprintf("image.tag=%s", tag),
+				}
+				chartImages, err := getImagesForChart(info)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get images for %s with error %w", info.name, err)
+				}
+				images = append(images, chartImages...)
+			}
+			// skip the to next addon because we got what we needed
+			continue
+		}
+		chartImages, err := getImagesForChart(info)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get images for %s with error %w", info.name, err)
+		}
+		images = append(images, chartImages...)
```

Comment:
I see there are some duplicate images being generated, what should be responsible for filtering those?
```
...
docker.io/nutanix/ntnx-csi:3.0.0
registry.k8s.io/sig-storage/livenessprobe:v2.11.0
registry.k8s.io/sig-storage/csi-provisioner:v3.6.3
registry.k8s.io/sig-storage/csi-attacher:v4.4.3
registry.k8s.io/sig-storage/csi-resizer:v1.9.3
registry.k8s.io/sig-storage/csi-snapshotter:v3.0.3
docker.io/nutanix/ntnx-csi:3.0.0
...
quay.io/metallb/speaker:v0.14.8
quay.io/frrouting/frr:9.1.0
quay.io/metallb/speaker:v0.14.8
...

Reply:
updated and it should remove duplicates